### PR TITLE
[4.3] Fix drag & drop extension install

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue [#39969](https://github.com/joomla/joomla-cms/issues/39969) .

### Summary of Changes
Fixes package installs. Very unsure about this. In the frontend not sure we can really guarentee urls will looks like this (they might be sef'd up and not starting with a slash or index.php) we might well be better off reverting the original // cc @dgrammatiko 

### Testing Instructions
See linked issue

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/media_source/system/js/core.es6.js [hunk 1]
```diff
@@ -623,8 +623,9 @@ window.Joomla.Modal = window.Joomla.Modal || {
         const token = Joomla.getOptions('csrf.token', '');
 
         // Use the CSRF only on the site's domain
-        // eslint-disable-next-line no-restricted-globals
-        if (token && (newOptions.url.startsWith('/') || newOptions.url.startsWith(location.origin))) {
+        if (token
+          // eslint-disable-next-line no-restricted-globals
+          && (newOptions.url.startsWith('index.php') || newOptions.url.startsWith('/') || newOptions.url.startsWith(location.origin))) {
```
Comment (dgrammatiko) [inline build/media_source/system/js/core.es6.js]:
```suggestion
          && (newOptions.url.startsWith('index.php?') || newOptions.url.startsWith('/') || newOptions.url.startsWith(location.origin))) {
```

**Review by dgrammatiko — APPROVED** [review]

**Comment by brianteeman** [comment]
Sorry but this doesnt fix the reported issue

**Comment by Fedik** [comment]
hmhm, then better check for `http`, kind of:
```
(
  token
&&
 (
  !(newOptions.url.startsWith('http:') || newOptions.url.startsWith('https:'))
  ||
  newOptions.url.startsWith(location.origin)
 )
)
```
(If the url does not have http(s), or it startsWith(location.origin))
Need to test.

**Comment by Fedik** [comment]
The thing with `http` works for me locally. @wilsonge do you update PR or I do new? :wink:

**Comment by wilsonge** [comment]
Just make it :)

**Comment by Fedik** [comment]
Alternative PR #40080

**Comment by obuisard** [comment]
Close in favor of #40080. Thank you George @wilsonge !

**Pull request closed without merging**